tests/brevitas/nn/test_recurrent.py

```diff
@@ -87,6 +88,11 @@ def test_rnn_quant_disabled_fwd(
         assert torch.isclose(out[0], ref_out[0], atol=ATOL).all()
         assert torch.isclose(out[1], ref_out[1], atol=ATOL).all()
 
+        # Test that the brevitas model can be saved/loaded without warning
+        with warnings.catch_warnings(record=True) as wlist:
```

Comment:
Generally speaking, I prefer having a separate function for each aspect we want to test, even at the cost of duplicating code (which could be prevented with fixtures and parametrization).

Keeping things simple, for this assert my understanding is that we don't need to parametrize the different variations of RNN/LSTM (but please correct me if I'm wrong). Definitely we don't need to run it for every input tensor generated by hypothesis.

I would create two separate tests with a single version of RNN/LSTM, and test this assert there.

In a separate PR we can figure out a way to leverage pytest_cases to test different implementations of RNN/LSTMs for all tests here.

Another option would be to leave it like this for the moment, and address everything in a separate PR. What's your opinion @volcacius?